util/ranger/detacher.go

```diff
@@ -240,7 +237,21 @@ func extractIndexPointRangesForCNF(sctx sessionctx.Context, conds []expression.E
 	if r != nil {
 		r.IsDNFCond = false
 	}
-	return r, offset, nil
+	return r, offset, columnValues, nil
+}
+
+func unionColumnValues(lhs, rhs []*valueInfo, numCols int) []*valueInfo {
+	if lhs == nil {
+		lhs = make([]*valueInfo, numCols)
+	}
+	if rhs != nil {
+		for i, valInfo := range lhs {
+			if valInfo == nil && rhs[i] != nil {
+				lhs[i] = rhs[i]
+			}
+		}
+	}
```

Comment:
When both `lhs` and `rhs` are `nil`, it will return a slice of `nil`. Is this expected?

Reply:
Yes, a slice of `nil` means no column is constant. But we can just return `nil` if both `lhs` and `rhs` are `nil`, which avoids `make`.